2996_reflection/reflection.md

```diff
@@ -4423,50 +4456,64 @@ consteval vector<info> accessible_subobjects_of(info target, info from);
 ::: std
 ::: addu
 ```cpp
-consteval size_t offset_of(info entity);
-consteval size_t size_of(info entity);
-consteval size_t alignment_of(info entity);
-consteval size_t bit_offset_of(info entity);
-consteval size_t bit_size_of(info entity);
+consteval size_t offset_of(info r);
 ```
-:::
-:::
 
-### [meta.reflection.result] Expression result reflection {-}
+[#]{.pnum} *Mandates*: `r` is a reflection designating a non-static data member or base class.
+
+[#]{.pnum} *Returns*: The offset in bytes, from the beginning of any object of type `parent_of(r)`, to the subobject associated with the entity reflected by `r`.
 
-::: std
-::: addu
 ```cpp
-template <typename T>
-  consteval info reflect_value(T expr);
+consteval size_t size_of(info r);
 ```
 
-[#]{.pnum} *Mandates*: `T` is a structural type. `T` is not a reference type. Any subobject of the value computed by `expr` having reference or pointer type designates an entity that is a permitted result of a constant expression.
+[#]{.pnum} *Mandates*: `r` is a reflection of a type, non-static data member, base class, object, or variable.
 
-[#]{.pnum} *Returns*: A reflection of the value computed by an lvalue-to-rvalue conversion applied to `expr`. The type of the reflected value is the cv-unqualified version of `T`.
+[#]{.pnum} *Returns* If `r` designates a type `T`, then `sizeof(T)`. Otherwise, `size_of(type_of(r))`.
 
 ```cpp
-template <typename T>
-  consteval info reflect_object(T& expr);
+consteval size_t alignment_of(info r);
 ```
 
-[#]{.pnum} *Mandates*: `T` is not a function type. `expr` designates an entity that is a permitted result of a constant expression.
+[#]{.pnum} *Mandates*: `r` is a reflection designating an object, variable, type, non-static data member, or base class.
 
-[#]{.pnum} *Returns*: A reflection of the object referenced by `expr`. If the reflected object is a variable `Obj`, the returned reflection compares equal to `^Obj`.
+[#]{.pnum} *Returns*: If `r` designates a type, object, or variable, then the alignment requirement of the entity. Otherwise, if `r` designates a base class, then `alignment_of(type_of(r))`. Otherwise, the alignment requirement of the subobject associated with the reflected non-static data member within any object of type `parent_of(r)`.
 
 ```cpp
-template <typename T>
-  consteval info reflect_function(T& expr);
+consteval size_t bit_offset_of(info r);
 ```
 
-[#]{.pnum} *Mandates*: `T` is a function type.
+[#]{.pnum} *Mandates*: `r` is a reflection designating a non-static data member.
 
-[#]{.pnum} *Returns*: `^fn`, where `fn` is the function referenced by `expr`.
+[#]{.pnum} *Returns*: The offset in bits, from the beginning of any object of type `parent_of(r)`, to the subobject associated with the entity reflected by `r`, minus the value of `offset_of(r);`
```

Comment:
I'd rather write this as:

```suggestion
[#]{.pnum} Let `V` be the offset in bits from the beginning of an object of type `parent_of(r)` to the subobject associated with the entity reflected by `r`.

[#]{.pnum} *Returns*: `V - offset_of(r)`.
```

Also fixes a typo on any --> an.

I think this makes it more obvious that there's a subtraction going on, otherwise the "minus" is kind of far into the words.

Reply:
Done.